[KEYCLOAK-12905] [KEYCLOAK-12920] Fixes

iankko opened the pull request:
<hr/>

    [KEYCLOAK-12905] PostgreSQL templates: Update PostgreSQL image stream tag version to "10"
    
    Utilize PostgreSQL of version v10 by default, when deploying the Red Hat Single Sign-On 7.3
    for OpenShift image from PostgreSQL templates
    
    Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

<hr/>

    [KEYCLOAK-12920] Persistent templates: Adjust "timeoutSeconds" and "initialDelaySeconds"
    to prevent readiness / liveness probe to end up DB pod lifecycle prematurely
    (yet before the DB server has had chance properly to initialize)
    
    Customize:
    * "timeoutSeconds" to 10 and
    * "initialDelaySeconds" to 90
    
    and also specify:
    * "successThreshold:" to 1 and
    * "failureThreshold" to 3
    
    on readiness and liveness probes in persistent templates to avoid readiness / liveness probe
    to bail out DB pod lifecycle prematurely due to:
    * "Inappropriate ioctl for device" event (case of MySQL probes), or
    * Issues like https://github.com/sclorg/postgresql-container/issues/309 (case of PostgreSQL probes)
    
    Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

<hr/>

Thanks for submitting your Pull Request!

Please make sure your PR meets the following requirements:

- [ ] Pull Request title is properly formatted: `[CLOUD-XYA] Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket
- [ ] Attached commits represent units of work and are properly formatted
- [ ] You have read and agreed to the Developer Certificate of Origin (DCO) (see `CONTRIBUTING.md`)
- [ ] Every commit contains `Signed-off-by: Your Name <yourname@example.com>` - use `git commit -s`

Commit:
[KEYCLOAK-12905] PostgreSQL templates: Update PostgreSQL image stream tag version to "10"

Utilize PostgreSQL of version v10 by default, when deploying the Red Hat Single Sign-On 7.3
for OpenShift image from PostgreSQL templates

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

iankko commented:
__Example verification steps for _KEYCLOAK-12905_ problem:__

To verify the updated template still works (as expected), you can test e.g. as follows (steps below assume the `redhat-sso-7-openshift-image` is the current working directory):

They will create updated / overriden version of the `sso73-x509-postgresql-persistent` in the current project (compared to that one already present in the `openshift` namespace).

```
$ oc login [OpenShift console URL] -u [username] -p [password]
$ export PROJECT="keycloak-12905-postgresql-10-demo"
$ oc new-project $PROJECT
$ cd templates/
$ oc get templates
No resources found.
$ oc create -f sso73-x509-postgresql-persistent.json 
template.template.openshift.io/sso73-x509-postgresql-persistent created
$ oc get templates
NAME                               DESCRIPTION                                                                       PARAMETERS     OBJECTS
sso73-x509-postgresql-persistent   An example application based on RH-SSO 7.3 image. For more information about...   21 (9 blank)   7
$ oc new-app --template=sso73-x509-postgresql-persistent
...
--> Success
...
    Run 'oc status' to view your app.
$ oc get pods
NAME                     READY     STATUS    RESTARTS   AGE
sso-1-fmvfc              1/1       Running   1          7m
sso-postgresql-1-5vzhq   1/1       Running   0          7m
$ oc rsh sso-postgresql-1-5vzhq
sh-4.2$ which psql
/opt/rh/rh-postgresql10/root/usr/bin/psql
sh-4.2$ psql --version
psql (PostgreSQL) 10.6
sh-4.2$ exit      
```

iankko commented:
@hmlnarik @pepo48 PTAL once got a chance

@pskopek @sguilhen jFYI

Thanks, Jan

hmlnarik reviewed (approved):
Looks good from my POV. Thanks @iankko

Commit:
[KEYCLOAK-12920] Persistent templates: Adjust "timeoutSeconds" and "initialDelaySeconds"
to prevent readiness / liveness probe to end up DB pod lifecycle prematurely
(yet before the DB server has had chance properly to initialize)

Customize:
* "timeoutSeconds" to 10 and
* "initialDelaySeconds" to 90

and also specify:
* "successThreshold:" to 1 and
* "failureThreshold" to 3

on readiness and liveness probes in persistent templates to avoid readiness / liveness probe
to bail out DB pod lifecycle prematurely due to:
* "Inappropriate ioctl for device" event (case of MySQL probes), or
* Issues like https://github.com/sclorg/postgresql-container/issues/309 (case of PostgreSQL probes)

Signed-off-by: Jan Lieskovsky <jlieskov@redhat.com>

iankko commented:
__Example verification steps for _KEYCLOAK-12920_ problem:__

Deploy the updated template similarly like in KEYCLOAK-12905 case above. With the change there shouldn't be _"Readiness probe failed: Inappropriate ioct for device"_ message, when deploying (some of) the persistent templates. The RH-SSO pod should still deploy successfully.

hmlnarik commented:
The second commit looks also good, thank you!

Pepo48 commented:
I've deployed PostgreSQL using the provided template - everything worked as expected. Both commits LGTM. Thanks Jan!

iankko commented:
@hmlnarik @Pepo48 Great. Thanks to both for review!
